molgenis-scripts/src/main/java/org/molgenis/script/ScriptRunnerController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import static java.text.MessageFormat.format;
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.support.Href.concatEntityHref;
+
+/**
+ * Controller for running a script
+ * <p>
+ * Url: /scripts/${scriptname}/run
+ * <p>
+ * If the script generates an outputfile that file is streamed to the outputstream else the script output
+ */
+@Controller
+public class ScriptRunnerController
+{
+	private final ScriptJobExecutionFactory scriptJobExecutionFactory;
+	private final JobExecutor jobExecutor;
+	private final SavedScriptRunner savedScriptRunner;
+	private final Gson gson;
+	private final ExecutorService executorService;
+	private final MenuReaderService menuReaderService;
+	private final DataService dataService;
+	private final UserAccountService userAccountService;
+
+	private ScriptRunnerController(ScriptJobExecutionFactory scriptJobExecutionFactory, JobExecutor jobExecutor,
+			SavedScriptRunner savedScriptRunner, Gson gson, ExecutorService executorService,
+			MenuReaderService menuReaderService, DataService dataService, UserAccountService userAccountService)
+	{
+		this.scriptJobExecutionFactory = requireNonNull(scriptJobExecutionFactory);
+		this.jobExecutor = requireNonNull(jobExecutor);
+		this.savedScriptRunner = requireNonNull(savedScriptRunner);
+		this.gson = requireNonNull(gson);
+		this.executorService = requireNonNull(executorService);
+		this.menuReaderService = requireNonNull(menuReaderService);
+		this.dataService = requireNonNull(dataService);
+		this.userAccountService = requireNonNull(userAccountService);
+	}
+
+	@RequestMapping("/scripts/{name}/start")
+	public void startScript(@PathVariable("name") String scriptName, @RequestParam Map<String, Object> parameters,
+			HttpServletResponse response) throws IOException, URISyntaxException
+	{
+		ScriptJobExecution scriptJobExecution = scriptJobExecutionFactory.create();
+		scriptJobExecution.setName(scriptName);
+		scriptJobExecution.setParameters(gson.toJson(parameters));
+		scriptJobExecution.setUser(userAccountService.getCurrentUser().getUsername());
+
+		dataService.add(scriptJobExecution.getEntityType().getId(), scriptJobExecution);
+
+		executorService.submit(() -> jobExecutor.execute(scriptJobExecution));
+
+		String jobHref = concatEntityHref("/api/v2", scriptJobExecution.getEntityType().getId(),
+				scriptJobExecution.getIdValue());
+		String jobControllerURL = menuReaderService.getMenu().findMenuItemPath(JobsController.ID);
+		response.sendRedirect(format("{0}/viewJob/?jobHref={1}&refreshTimeoutMillis=1000", jobControllerURL, jobHref));
```

Comment:
Should we make some sort of utility or shared string for creating this complicated url?